deploy/aws/java11Exec/src/main/java/poseidon/App.java

```diff
@@ -0,0 +1,136 @@
+                .build();
+        this.connectionID = ctx.getConnectionId();
+        this.disableOutput = input.getHeaders() != null && input.getHeaders().containsKey(disableOutputHeaderKey) && Boolean.parseBoolean(input.getHeaders().get(disableOutputHeaderKey));
+        AwsFunctionRequest execution = gson.fromJson(input.getBody(), AwsFunctionRequest.class);
+
+        try {
+            File workingDirectory = this.writeFS(execution.files);
+
+            ProcessBuilder pb = new ProcessBuilder(execution.cmd).redirectErrorStream(true);
+            pb.directory(workingDirectory);
+            Process p = pb.start();
+            InputStream stdout = p.getInputStream(), stderr = p.getErrorStream();
+            this.forwardOutput(p, stdout, stderr);
+            p.destroy();
+            return new APIGatewayProxyResponseEvent().withStatusCode(200);
+        } catch (Exception e) {
+            this.sendMessage(e.toString().getBytes(StandardCharsets.UTF_8), WebSocketMessageType.WebSocketOutputError, 0);
+            return new APIGatewayProxyResponseEvent().withBody(e.toString()).withStatusCode(500);
+        }
+    }
+
+    // writeFS writes the files to the local filesystem.
+    private File writeFS(Map<String, String> files) throws IOException {
+        File workspace = Files.createTempDirectory("workspace").toFile();
+        for (Map.Entry<String, String> entry : files.entrySet()) {
+            File f = new File(workspace, entry.getKey());
+
+            boolean ok = f.getParentFile().mkdirs();
+            if (!ok) throw new IOException("Cannot create parent directories.");
+
+            ok = f.createNewFile();
+            if (!ok) throw new IOException("Cannot create file.");
+
+            Files.write(f.toPath(), Base64.getDecoder().decode(entry.getValue()));
+        }
+        return workspace;
+    }
+
+    // forwardOutput sends the output of the process to the WebSocket connection.
+    private void forwardOutput(Process p, InputStream stdout, InputStream stderr) throws InterruptedException, IOException {
+        while (p.isAlive() || stdout.available()>0 || stderr.available()>0) {
+            while(stdout.available()>0) this.sendMessage(stdout.readAllBytes(), WebSocketMessageType.WebSocketOutputStdout, 0);
+            while(stderr.available()>0) this.sendMessage(stderr.readAllBytes(), WebSocketMessageType.WebSocketOutputStderr, 0);
+            Thread.sleep(50);
+        }
+        this.sendMessage(new byte[0], WebSocketMessageType.WebSocketExit, p.exitValue());
+    }
+
+    // sendMessage sends WebSocketMessage objects back to the requester of this Lambda function.
+    private void sendMessage(byte[] data, WebSocketMessageType type, int exitCode) {
+        if (this.disableOutput) return;
+        WebSocketMessage msg = new WebSocketMessage();
+        msg.data = new String(data);
+        msg.type = type;
+        msg.exitCode = exitCode;
```

Comment:
Does that mean every message contains a (faked) exit code even if does not represent a process termination but is part of an output feed? 

And should those message adhere to Poseidon's web socket scheme? This one does not specify the `exitCode` property and simultaneously sets `additionalProperties` to `false`.

Reply:
I have oriented myself at the dto.WebSocketMessage but the WebSocket Schema is the right reference point. Thanks!